app/src/main/java/org/systers/mentorship/view/activities/FeedbackActivity.kt

```diff
@@ -0,0 +1,216 @@
+                    feedbackEmailErr = false
+                }
+                }
+            }
+
+        })
+
+        //init message input field
+        FeedbackPageMessage.isErrorEnabled = true
+        FeedbackPageMessage.error = "Message can't be left blank"
+        FeedbackPageMessage.editText?.addTextChangedListener(object : TextWatcher{
+            override fun afterTextChanged(s: Editable?) {
+
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                if (s.toString().length > FeedbackPageMessage.counterMaxLength)
+                {
+                    FeedbackPageMessage.isErrorEnabled = true
+                    FeedbackPageMessage.error = "Please don't exceed the character limit"
+                    feedbackMsgErr = true
+                }
+                else
+                {
+                    if (s.toString().isEmpty())
+                    {
+                        FeedbackPageMessage.isErrorEnabled = true
+                        FeedbackPageMessage.error = "Message can't be left blank"
+                        feedbackMsgErr = true
+                    }
+                    else
+                    {
+                        FeedbackPageMessage.isErrorEnabled = false
+                        feedbackMsgErr = false
+                    }
+                }
+            }
+        })
+
+        //init ratingBtnList, an arraylist of ImageButtons, it is going to be used to add rating and change image resource, see fun assignRating()
+        ratingBtnList = ArrayList()
+        ratingBtnList?.add(FeedbackpageS1)
+        ratingBtnList?.add(FeedbackpageS2)
+        ratingBtnList?.add(FeedbackpageS3)
+        ratingBtnList?.add(FeedbackpageS4)
+        ratingBtnList?.add(FeedbackpageS5)
+        //set the onclick listener for each of the star buttons
+        for(i in ratingBtnList!!)
+        {
+            i.setOnClickListener(this)
+        }
+
+        //set click listeners for category radio buttons
+        FeedbackpageRd1.setOnClickListener(this)
+        FeedbackpageRd2.setOnClickListener(this)
+        FeedbackpageRd3.setOnClickListener(this)
+        //init category = "bug"
+        feedbackCategory = "Bug"
+
+        //Final submit button
+        FeedbackpageSendbtn.setOnClickListener {
+            if (!feedbackEmailErr && !feedbackMsgErr && feedbackRating != 0) {
+                Toast.makeText(this, "Thank you for your feedback!", Toast.LENGTH_SHORT).show()
+                //add backend code for adding rating, category, message, email ID
+            }
+            else
+            {
+                if (feedbackMsgErr || feedbackEmailErr || feedbackRating == 0) {
+                    Toast.makeText(this, "Please fill in the input fields and/or assign a rating", Toast.LENGTH_SHORT).show()
```

Comment:
should be in `strings.xml` file, in this way, we cannot have translations on this string